pkgs/tools/backup/mydumper/default.nix

```diff
@@ -1,43 +1,81 @@
 { lib, stdenv, fetchFromGitHub
 , cmake, pkg-config, sphinx
-, glib , pcre
+, glib, pcre, pcre2
 , libmysqlclient, libressl
-, zlib, zstd
-}:
+, zlib, zstd, nix-update-script, util-linux
+, libselinux, libsepol
+, testers, versionCheckHook, mydumper }:
 
 stdenv.mkDerivation rec {
   pname = "mydumper";
-  version = "0.14.3-1";
+  version = "0.16.7-5";
 
   src = fetchFromGitHub {
-    owner  = "mydumper";
-    repo = "mydumper";
+    owner = pname;
+    repo = pname;
     rev = "refs/tags/v${version}";
-    hash = "sha256-qyJGnrBOElQ3s2VoOWfW1luacd33haanmzKidMBgCpc=";
+    hash = "sha256-DxH2ZP7BSnBfiPBKi1ZBlXNhJeXMzt+V/FMy2nK3nAA=";
+    # as of mydumper v0.16.5-1, mydumper extracted its docs into a submodule
+    fetchSubmodules = true;
   };
 
   outputs = [ "out" "doc" "man" ];
 
   nativeBuildInputs = [ cmake pkg-config sphinx ];
 
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  doInstallCheck = true;
+
   buildInputs = [
-    glib pcre
+    glib pcre pcre2
     libmysqlclient libressl
-    zlib zstd
-  ];
+    zlib zstd util-linux
+  ] ++ lib.optionals stdenv.isLinux [ libselinux libsepol ];
 
   cmakeFlags = [
     "-DCMAKE_SKIP_BUILD_RPATH=ON"
     "-DMYSQL_INCLUDE_DIR=${lib.getDev libmysqlclient}/include/mysql"
     "-DWITH_ZSTD=ON"
```

Comment:
This `     "-DWITH_ZSTD=ON"` can be removed, it is not used anymore

Reply:
(for context: David is mydumper's author)